src/kv/kv.h

```diff
@@ -1614,14 +1615,21 @@ namespace kv
             last_replicated + offset, std::move(data_), committable_);
           pending_txs.erase(search);
         }
+
+        if (batch.size() == 0)
+          return CommitSuccess::OK;
+
+        previous_rollback_count = rollback_count;
         previous_last_replicated = last_replicated;
         next_last_replicated = last_replicated + batch.size();
       }
 
       if (r->replicate(batch))
       {
         std::lock_guard<SpinLock> vguard(version_lock);
-        if (last_replicated == previous_last_replicated)
+        if (
```

Comment:
If last_replicated has not changed, but a rollback occured since we attempted to replicate, we do not want to move last_replicated forward.